### PR TITLE
Fix custom-lib race

### PR DESCRIPTION
**Pull request by lizmat**
As described in #4737

### CODE REVIEW

File: src/core.c/CompUnit/RepositoryRegistry.pm6 [hunk 1]
```diff
@@ -53,6 +48,21 @@ class CompUnit::RepositoryRegistry {
         }
     }
 
+    method !register-custom-lib-repo(
```
Comment (ugexe) [inline src/core.c/CompUnit/RepositoryRegistry.pm6:51]: For consistency this should be `method !register-custom-lib-repository`, even if it is a private method.

Comment (lizmat) [reply src/core.c/CompUnit/RepositoryRegistry.pm6]: Addressed in https://github.com/rakudo/rakudo/pull/4738/commits/8e034779d41b65b75fcad7c497906e085b48fa15

File: src/core.c/CompUnit/RepositoryRegistry.pm6 [hunk 2]
```diff
@@ -255,19 +258,27 @@ class CompUnit::RepositoryRegistry {
 
     method repository-for-name(str $name) {
         $*REPO; # initialize if not yet done
-        nqp::ifnull(nqp::atkey($custom-lib,$name),Nil)
+        $lock.protect: {
+            nqp::ifnull(nqp::atkey($custom-lib,$name),Nil)
+        }
     }
 
+    # this feels like dead code  XXX
```
Comment (ugexe) [inline src/core.c/CompUnit/RepositoryRegistry.pm6:266]: This comment is mistaken. For example -- https://github.com/ugexe/CompUnit--Repository--Tar/blob/47021114ba3dbfd9c44c307d300853d7fc773fab/lib/CompUnit/Repository/Tar.rakumod#L16

Comment (lizmat) [reply src/core.c/CompUnit/RepositoryRegistry.pm6]: Addressed in https://github.com/rakudo/rakudo/commit/f52e4b995c

**Review by ugexe — APPROVED** [review]

**Pull request merged**